core/node/node_framework/src/implementations/layers/metadata_calculator.rs

```diff
@@ -154,3 +170,14 @@ impl Task for TreeApiTask {
             .await
     }
 }
+
+#[async_trait::async_trait]
+impl Task for MerkleTreePruningTask {
+    fn id(&self) -> TaskId {
+        "merkle_tree_pruning_task".into()
+    }
```

Comment:
nit: we don't mention `task` in other implementations of `fn id(&self) -> TaskId {`

Reply:
IMO `merkle_tree_pruning` would sound weird. E.g. here it's a task not because it's `impl Task for <...>`, but it's _really_ a part of its name.